pkg/validation/internal/operatorhub.go

```diff
@@ -17,51 +18,72 @@ import (
 	interfaces "github.com/operator-framework/api/pkg/validation/interfaces"
 )
 
+const minKubeVersionWarnMessage = "csv.Spec.minKubeVersion is not informed. It is recommended you provide this information. " +
+	"Otherwise, it would mean that your operator project can be distributed and installed in any cluster version " +
+	"available, which is not necessarily the case for all projects."
+
+const crdv1beta1DeprecationMsg = "apiextensions.k8s.io/v1beta1, kind=CustomResourceDefinitions was deprecated in " +
+	"Kubernetes v1.16 and will be removed in v1.22 in favor of v1"
+
 var OperatorHubValidator interfaces.Validator = interfaces.ValidatorFunc(validateOperatorHub)
 
 var validCapabilities = map[string]struct{}{
-	"Basic Install":     struct{}{},
-	"Seamless Upgrades": struct{}{},
-	"Full Lifecycle":    struct{}{},
-	"Deep Insights":     struct{}{},
-	"Auto Pilot":        struct{}{},
+	"Basic Install":     {},
+	"Seamless Upgrades": {},
+	"Full Lifecycle":    {},
+	"Deep Insights":     {},
+	"Auto Pilot":        {},
 }
 
 var validMediatypes = map[string]struct{}{
-	"image/gif":     struct{}{},
-	"image/jpeg":    struct{}{},
-	"image/png":     struct{}{},
-	"image/svg+xml": struct{}{},
+	"image/gif":     {},
+	"image/jpeg":    {},
+	"image/png":     {},
+	"image/svg+xml": {},
 }
 
 var validCategories = map[string]struct{}{
-	"AI/Machine Learning":    struct{}{},
-	"Application Runtime":    struct{}{},
-	"Big Data":               struct{}{},
-	"Cloud Provider":         struct{}{},
-	"Developer Tools":        struct{}{},
-	"Database":               struct{}{},
-	"Integration & Delivery": struct{}{},
-	"Logging & Tracing":      struct{}{},
-	"Monitoring":             struct{}{},
-	"Networking":             struct{}{},
-	"OpenShift Optional":     struct{}{},
-	"Security":               struct{}{},
-	"Storage":                struct{}{},
-	"Streaming & Messaging":  struct{}{},
+	"AI/Machine Learning":    {},
+	"Application Runtime":    {},
+	"Big Data":               {},
+	"Cloud Provider":         {},
+	"Developer Tools":        {},
+	"Database":               {},
+	"Integration & Delivery": {},
+	"Logging & Tracing":      {},
+	"Monitoring":             {},
+	"Networking":             {},
+	"OpenShift Optional":     {},
+	"Security":               {},
+	"Storage":                {},
+	"Streaming & Messaging":  {},
 }
 
 func validateOperatorHub(objs ...interface{}) (results []errors.ManifestResult) {
+
+	// Obtain the k8s version if informed via the objects an optional
+	k8sVersion := ""
+	for _, obj := range objs {
+		switch obj.(type) {
+		case map[string]string:
+			k8sVersion = obj.(map[string]string)["k8s"]
```

Comment:
It would be nice to define something like terraform’s [`TF_VARS_name`](https://www.terraform.io/docs/cli/config/environment-variables.html) for validators, where the prefix is the validator name capitalized, so you don’t have to pass in some object or define flags for the validator lib consumer.

Just an idea for a follow up, this is fine for now.

Reply:
I am not sure about that. Why should we adopt a Terraform convention?

Reply:
My suggestion is not a terraform convention, just an idea lifted from terraform.

Reply:
IHMO I do not think that would be good to have a prefix here. 
The idea is we are able to use the same key for 1 or Many validators since its usage via SDK can be done via the suite.